0MI0800304/7.03/ConsoleApplication1/problem123.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <fstream>
+//problem 1
+void writeData() {
+	std::ofstream file("result.txt");
+	if (!file.is_open()) {
+		std::cout << "Error!" << std::endl;
+		return;
+	}
+	int a = 0, b = 0, c = 0;
+	std::cin >> a >> b >> c;
+	file << a + b + c << " " << a * b * c << std::endl;
+	file.close();
+
+}
+void readData() {
+	std::ifstream file("result.txt");
+	if (!file.is_open()) {
+		std::cout << "Error!" << std::endl;
+		return;
+	}
+	int a = 0, b = 0;
+	file >> a >> b;
+	std::cout << a - b;
+	file.close();
+	}
+//problem 2
+int charCount() {
+	std::ifstream file("myFile.txt");
+	int count = 0;
+	if (!file.is_open())
+	{
+		std::cout << "Error!" <<std::endl;
+		return -1;
+	}
+	while (!file.eof()){
+		char ch;
+		file.get(ch);
+		if (ch != 'a' && ch != 'd')count++;
+	}
+	file.close();
+	return count;
+}
+//problem 3
+struct Pair {
+	unsigned x;
+	unsigned y;
+};
+Pair createPair(unsigned x,unsigned y) {
+	Pair pair;
+	pair.x = x;
+	pair.y = y;
+}
+void initPair(Pair& toInit, unsigned x, unsigned y) {
+	toInit.x = x;
+	toInit.y = y;
+}
+Pair readPairFromFile(std::ifstream& inFile) {
```

Comment:
Closing the file is best done in the same function where it was opened. Otherwise strange things happen - one gives you a file so that you can read something from it, you close the file without its creator knowing, then someone else may try reading from it which will end up rather bad :))